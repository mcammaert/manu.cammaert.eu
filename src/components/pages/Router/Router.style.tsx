import styled from 'styled-components';

export const Router = styled.div``;

export default { Router };
