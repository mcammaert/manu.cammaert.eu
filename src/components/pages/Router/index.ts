import * as S from './Router.style';

export { default as Router } from './Router';
export * from './Router.types';
export { S };
