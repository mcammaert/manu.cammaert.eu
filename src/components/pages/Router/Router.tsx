import React from 'react';

import { Switch, Route, useLocation } from 'react-router-dom';
import loadable from '@loadable/component';
import pMinDelay from 'p-min-delay';
import { useTransition, animated } from 'react-spring';

import { Header } from 'components/organisms/Header';
import { Loading as LoadingComponent } from 'components/molecules/Loading';
import { Container } from 'components/atoms/Container';
import { Error as ErrorPage } from 'components/pages/Error';

import { RouterProps } from './Router.types';

const fallbackTimeout = 20;

const Loading = () => (
  <Container>
    <LoadingComponent />
  </Container>
);

const Home = loadable(() => pMinDelay(import('components/pages/Home/Home'), fallbackTimeout), {
  fallback: <Loading />,
});
const Resume = loadable(() => pMinDelay(import('components/pages/Resume/Resume.container'), fallbackTimeout), {
  fallback: <Loading />,
});
const Contact = loadable(() => pMinDelay(import('components/pages/Contact/Contact'), fallbackTimeout), {
  fallback: <Loading />,
});
const PageNotFound = () => <ErrorPage message="Pagina niet gevonden." />;

const Router: React.FC<RouterProps> = () => {
  const location = useLocation();

  const transitions = useTransition(location, loc => loc.pathname, {
    from: { opacity: 0, transform: 'translate3d(-20px, 0, 0)' },
    enter: { opacity: 1, transform: 'translate3d(0, 0, 0)' },
    leave: { opacity: 0, transform: 'translate3d(-20px, 0, 0)' },
  });

  return (
    <>
      <Header />
      {transitions.map(({ item, props, key }) => (
        <animated.div key={key} style={props}>
          <Switch location={item}>
            <Route path="/cv" exact component={Resume} />
            <Route path="/contact" exact component={Contact} />
            <Route path="/" exact component={Home} />
            <Route path="*" component={PageNotFound} />
          </Switch>
        </animated.div>
      ))}
    </>
  );
};

export default Router;
