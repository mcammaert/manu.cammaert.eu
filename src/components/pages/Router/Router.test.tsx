import React from 'react';

import { BrowserRouter } from 'react-router-dom';
import { cleanup } from '@testing-library/react';
import 'jest-styled-components';

import { renderComponent } from 'testing/react-testing-library';

import Router from './Router';

afterEach(cleanup);

const theme = {
  colors: {},
};

describe('Router', () => {
  describe('default props', () => {
    it('should match the snapshot', () => {
      const component = (
        <BrowserRouter>
          <Router />
        </BrowserRouter>
      );
      const { container } = renderComponent({ theme, component });
      expect(container).toMatchSnapshot();
    });
  });
});
