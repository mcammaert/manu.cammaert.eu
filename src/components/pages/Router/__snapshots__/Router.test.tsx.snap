// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router default props should match the snapshot 1`] = `
.c0 {
  position: absolute;
}

<div>
  <div
    class=""
  >
    <div
      class=""
    >
      Header
      <ul
        class=""
      >
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/cv"
          >
            Curriculum Vitae
          </a>
        </li>
        <li>
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </div>
    <div
      style="opacity: 0; transform: translate3d(100px, 0, 0);"
    >
      <div
        class="c0"
      >
        <p>
          Laden...
        </p>
      </div>
    </div>
  </div>
</div>
`;
